mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go

```diff
@@ -33,5 +35,8 @@ func dataSourceGoogleStorageBucketRead(d *schema.ResourceData, meta interface{})
 	}
 	log.Printf("[DEBUG] Read bucket %v at location %v\n\n", res.Name, res.SelfLink)
 
-	return setStorageBucket(d, config, res, bucket, userAgent)
+	setStorageBucket(d, config, res, bucket, userAgent)
+
+	d.Set("project", strconv.Itoa(int(res.ProjectNumber)))
```

Comment:
Be careful here, you return the projectNumber and the documentation said that project is the projectId. Also in all data.source adn resource project represent always a project_id. You shuld convert hte projectNumber into a projectId

Reply:
Was thought of that. But bucket is global, and the project_id is not used anywhere. Here is the api endpoint

```
GET /storage/v1/b/hostname-bucket?alt=json&prettyPrint=false HTTP/1.1
Host: storage.googleapis.com
```

And api returns [projectNumber](https://cloud.google.com/storage/docs/json_api/v1/buckets)

I think in the resource & datasource itself, it should have used project_number to match the API's.

Reply:
Agree but when you create a bucket you need to specify a project (for only the billing purpose) even if it is global. Also if you don't specify the `project` in the provider configuration, the `data.google_storage_bucket` returns the correct project_id. So somebody already did the job but it is not consistent. 

In the case of the provider_google, it seems that they concentrate on using only project_id (almost everywhere, for at least the attribute `project` and used `project_number` in other case) 

So in this case, I will argue that project should be a `project_id` regardless of the google API

Reply:
The problem is we can't convert project # -> project ID within the provider.

We have a default project ID from the provider-level configuration, but we can't tell if that matches a given project number

Reply:
That said, I don't think we should set the project to a number at all. It is probably preferable to just not set this

Reply:
Hm, I guess we can & do this in setStorageBucket. I just don't really think anything is wrong here as the bucket name is a unique identifier, so when we have that the project isn't used during lookup.

Reply:
Yeah, looking at below lib, there isn't a query available that can be used to convert project # -> project ID 

https://pkg.go.dev/google.golang.org/api/cloudresourcemanager/v3